lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -2523,6 +2524,28 @@ bool SwiftLanguageRuntimeImpl::GetDynamicTypeAndAddress_IndirectEnumCase(
   }
 }
 
+void SwiftLanguageRuntimeImpl::DumpTyperef(
+    CompilerType type, TypeSystemSwiftTypeRef *module_holder,
+    SwiftASTContext *swift_ast_context, Stream *s) {
+  if (!s)
+    return;
+  const auto *typeref = GetTypeRef(type, module_holder, swift_ast_context);
+  if (!typeref)
+    return;
+  // Hacky: since dump takes a FILE *, create a temporary file, read into it,
```

Comment:
This is very ugly, but couldn't think of a better way of doing it...

Reply:
A cleaner way might be to open a pipe() and turn that into FILE * with fdopen(). But this implementation isn't much worse.

Reply:
We could also add a parallel API to the reflection library that doesn't use FILE*.